ddtrace/tracer/log.go

```diff
@@ -0,0 +1,109 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package tracer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"runtime"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+)
+
+const (
+	unknown = "unknown"
+)
+
+// startupInfo contains various information about the status of the tracer on startup.
+type startupInfo struct {
+	Date                  string            `json:"date"`                    // ISO 8601 date and time of start
+	OSName                string            `json:"os_name"`                 // Windows, Darwin, Debian, etc.
+	OSVersion             string            `json:"os_version"`              // Version of the OS
+	Version               string            `json:"version"`                 // Tracer version
+	Lang                  string            `json:"lang"`                    // "Go"
+	LangVersion           string            `json:"lang_version"`            // Go version, e.g. go1.13
+	Env                   string            `json:"env"`                     // Tracer env
+	Service               string            `json:"service"`                 // Tracer Service
+	AgentURL              string            `json:"agent_url"`               // The address of the agent
+	AgentError            string            `json:"agent_error"`             // Any error that occurred trying to connect to agent
+	Debug                 bool              `json:"debug"`                   // Whether debug mode is enabled
+	AnalyticsEnabled      bool              `json:"analytics_enabled"`       // True if there is a global analytics rate set
+	SampleRate            string            `json:"sample_rate"`             // The default sampling rate for the rules sampler
+	SamplingRules         []SamplingRule    `json:"sampling_rules"`          // Rules used by the rules sampler
+	SamplingRulesError    string            `json:"sampling_rules_error"`    // Any errors that occurred while parsing sampling rules
+	Tags                  map[string]string `json:"tags"`                    // Global tags
+	RuntimeMetricsEnabled bool              `json:"runtime_metrics_enabled"` // Whether or not runtime metrics are enabled
+	HealthMetricsEnabled  bool              `json:"health_metrics_enabled"`  // Whether or not health metrics are enabled
+	ApplicationVersion    string            `json:"dd_version"`              // Version of the user's application
+	Architecture          string            `json:"architecture"`            // Architecture of host machine
+	GlobalService         string            `json:"global_service"`          // Global service string. If not-nil should be same as Service. (#614)
+}
+
+// agentReachable determines whether or not the tracer is able to connect
+// to endpoint, which should be the full URL returned by transport.endpoint
```

Comment:
How do we know if the agent is reachable or not? Does a non-nil error mean it's not? The documentation sounds like we're expecting a `bool` return but instead it's an error.

Reply:
From the functional standpoint (aka readers of the logged message): `{ ..., "agent_error": null, ...}` means I was able to send a `[]` to the agent, while `{ ..., "agent_error": "something here explaining what's wrong", ...}` I was not able to send an empty payload `[]` to the agent within the configured timeout.

Reply:
Let's then add a sentence to the documentation. Something like:
```go
// The returned error, when not nil, specifies the reason why the agent was unreachable.
```

Reply:
I will change `agentReachable`, which seems to imply a true/false return, to `checkAgent`, and this suggestion to the documentation.